exercises/linked-list/src/linked_list.h

```diff
@@ -4,29 +4,28 @@
 #include <stdbool.h>
 
 typedef int ll_data_t;
+struct list;
 
-struct list_item;
-
-// constructs a new list of items
-struct list_item **new_list(void);
+// constructs a new (empty) list
+struct list *list_create(void);
 
 // checks if the list is empty
-bool is_list_empty(struct list_item **list);
+bool list_is_empty(const struct list *list);
 
 // inserts item at back of list
-bool push(struct list_item **list, ll_data_t item_data);
+struct list_node *list_push(struct list *list, ll_data_t item_data);
```

Comment:
Where is `struct list_node` forward declared? I'm confused as to how this compiles.

Reply:
It's legal because that function declaration also forward declares `struct list_node`. I changed it to `bool`.

Reply:
Cool, didn't know you could do that. Thanks!